src/server/saved_objects/routes/create.js

```diff
@@ -7,19 +7,22 @@ export const createCreateRoute = (prereqs) => {
     config: {
       pre: [prereqs.getSavedObjectsClient],
       validate: {
+        query: Joi.object().keys({
+          overwrite: Joi.boolean().default(false)
+        }),
         params: Joi.object().keys({
           type: Joi.string().required()
         }).required(),
         payload: Joi.object({
-          attributes: Joi.object().required()
+          attributes: Joi.object().required(),
+          id: Joi.string().allow(null)
```

Comment:
I feels like the url is the right place to put the id: `POST /api/saved_objects/{type}/{id}`